Don't silently ignore kwargs when projparams are specified

gberardinelli opened the pull request:
<!-- Feel free to remove check-list items aren't relevant to your change -->

 - [x] Closes #552
 - [x] Tests added
 - [ ] Fully documented, including `history.rst` for all changes and `api/*.rst` for new API

snowman2 commented:
@gberardinelli this looks great :+1:

Two questions for you:
1. Would you like to squash the commits yourself or do you mind if I do it on merge?
2. Would you mind adding a test in `test/crs/test_crs.py` that is identical to the one you added to test the `Proj` object in `test/test_proj.py` except that it would be a `CRS` object?

gberardinelli commented:
@snowman2, sorry for my slow responses.   Added a CRS test and squashed.

For whatever reason I'm getting failures from 4 existing tests in `test/crs/test_crs.py`, maybe related to my environment?

snowman2 commented:
Thanks @gberardinelli!
